### PR TITLE
encoder: add fps for VP8/VP9

### PR DESCRIPTION
**Pull request by wudping**
Just adding fps for VP8/VP9

to fix issue [#696](https://github.com/01org/libyami/issues/696)

Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

**Comment by wudping** [comment]
please help to review.

File: tests/encodeInputCapi.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ typedef void* EncodeOutputHandler;
 
 EncodeInputHandler createEncodeInput(const char * inputFileName, uint32_t fourcc, int width, int height);
 
-EncodeOutputHandler createEncodeOutput(const char * outputFileName, int width, int height);
+EncodeOutputHandler createEncodeOutput(const char* outputFileName, int width, int height, int fps = 0);
```
Comment (xuguangxin) [inline tests/encodeInputCapi.h]: better change this to 30

File: tests/encodeinput.h [hunk 2]
```diff
@@ -155,11 +155,11 @@ class EncodeOutputVPX : public EncodeOutput {
     virtual const char* getMimeType() = 0;
     virtual bool write(void* data, int size);
 protected:
-    virtual bool init(const char* outputFileName, int width , int height);
+    virtual bool init(const char* outputFileName, int width, int height, int fps = 0);
```
Comment (xuguangxin) [inline tests/encodeinput.h]: better 30

File: tests/encodeinput.h [hunk 3]
```diff
@@ -155,11 +155,11 @@ class EncodeOutputVPX : public EncodeOutput {
...
     uint32_t getFourcc() {return m_fourcc;};
     void setFourcc(uint32_t fourcc) {m_fourcc = fourcc;};
 private:
-    void getIVFFileHeader(uint8_t *header, int width, int height);
+    void getIVFFileHeader(uint8_t* header, int width, int height, int fps = 0);
```
Comment (xuguangxin) [inline tests/encodeinput.h]: better 30

File: tests/encodeInputCapi.h (hunk 1 above)
Comment (wudping) [inline tests/encodeInputCapi.h]: okay, Thanks, It's really better.

Comment (wudping) [inline tests/encodeInputCapi.h]: but how about that:
setUint32(header + 16, fps <= 0 ? 30, fps); // rate

File: tests/encodeinput.h (hunk 2 above)
Comment (wudping) [inline tests/encodeinput.h]: thanks @xuguangxin , but how about:
setUint32(header + 16, fps <= 0 ? 30 : fps); // rate

?

File: tests/encodeinput.h (hunk 3 above)
Comment (wudping) [inline tests/encodeinput.h]: thanks @xuguangxin , but how about:
setUint32(header + 16, fps <= 0 ? 30 : fps); // rate

?

File: tests/encodeInputCapi.h (hunk 1 above)
Comment (xuguangxin) [inline tests/encodeInputCapi.h]: 0 is not a valid fps, right? so why not use a valid fps, why we need 0

Comment (wudping) [inline tests/encodeInputCapi.h]: I agree with you to modify my code to set a valid fps.
But I think I should replace
setUint32(header + 16, fps); // rate
with
setUint32(header + 16, fps <= 0 ? 30 : fps); // rate
That will just modify on one place.
But if I modify the format parameter fps = 30, there will be many places need to change.

**Comment by wudping** [comment]
updated with "setUint32(header + 16, fps <= 0 ? 30 : fps); // rate" to set the default value of fps.
please help to review.

**Comment by wudping** [comment]
updated, please help to review.

**Comment by wang-fei** [comment]
verify+1

**Comment by xuguangxin** [comment]
merged as 0b024ad25c8f9972dd8970642aa5c71fd70ad1c6

**Pull request closed without merging**